omod-common/src/main/java/org/openmrs/module/webservices/helper/TaskServiceWrapper.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.webservices.helper;
+
+import org.openmrs.scheduler.TaskDefinition;
+import org.openmrs.scheduler.SchedulerService;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.api.context.Context;
+
+import java.util.Collection;
+
+public class TaskServiceWrapper {
+	
+	public TaskDefinition getTaskById(String id) {
+		return Context.getSchedulerService().getTask(Integer.parseInt(id));
+	}
+	
+	public Collection<TaskDefinition> getScheduledTasks() {
+		return Context.getSchedulerService().getScheduledTasks();
+	}
+	
+	public Collection<TaskDefinition> getRegisteredTasks() {
+		return Context.getSchedulerService().getRegisteredTasks();
+	}
+	
+	public boolean isTaskStarted(TaskDefinition task) {
+		return task.getStarted();
+	}
+	
+	public boolean isTaskStopped(TaskDefinition task) {
+		return !isTaskStarted(task);
+	}
+	
+	public void saveTaskDefinition(TaskDefinition task) {
+		Context.getSchedulerService().saveTaskDefinition(task);
+	}
+	
+	public void unloadTask(TaskDefinition task) throws SchedulerException {
+		if (Context.getSchedulerService().getTaskByName(task.getName()).getStarted()) {
+			Context.getSchedulerService().shutdownTask(task);
+		}
+		Context.getSchedulerService().deleteTask(task.getId());
+	}
+	
+	public void scheduleTask(TaskDefinition task) throws SchedulerException {
+		if (!Context.getSchedulerService().getTaskByName(task.getName()).getStarted()) {
```

Comment:
After taking a second look, i think we do not need this call at all. All those check of ensure that a service is not started before schedule, etc, belong to the service layer API.
Your role is to simply call Context.getSchedulerService().scheduleTask(task) and that is all.

Reply:
Okay :-). I agree with you
No need of much conditions at this point